staging/src/k8s.io/api/batch/v1/types.go

```diff
@@ -35,6 +35,10 @@ const (
 	// apiserver and job controller will ignore this annotation and they will
 	// always track jobs using finalizers.
 	JobTrackingFinalizer = "batch.kubernetes.io/job-tracking"
+	// The Job labels will use batch.kubernetes.io as a prefix for all labels
+	// Currently the job controller will set job-name and controller-uid with this label and also sets the job-name
```

Comment:
Watch out for “Currently” in code; this can end up ageing. It's better to document behavior without that word, and maybe also write something like “Historically, the Job controller used the unprefixed labels (a bug), and Kubernetes continues to recognize those unprefixed labels for compatbility”.